### PR TITLE
Add project skill lists to UI

### PR DESCRIPTION
**Pull request by JoshSmith**
No tests really. Fixes some things with project skills.

### CODE REVIEW

File: app/components/skill-list-item.js [hunk 1]
```diff
@@ -4,6 +4,19 @@ export default Ember.Component.extend({
...
+  userSkills: Ember.inject.service(),
+  user: Ember.computed.alias('currentUser.user'),
+  usersUserSkills: Ember.computed.alias('user.userSkills'),
+
+  userSkill: Ember.computed('skill', 'userSkills.userSkills.isFulfilled', function() {
+    let skill = this.get('skill');
+    let userSkills = this.get('userSkills');
+    let result = userSkills.findUserSkill(skill);
+    skill.set('matched', result);
+    return result;
```
Comment (JoshSmith) [inline app/components/skill-list-item.js]: I don't really like the way I'm setting the virtual attr `matched` here. Would prefer a better way. This is not really data down.

File: app/routes/projects-list.js [hunk 2]
```diff
@@ -1,6 +1,14 @@
 import Ember from 'ember';
 
 export default Ember.Route.extend({
+  currentUser: Ember.inject.service(),
+
+  beforeModel() {
+    this._super(...arguments);
+    let user = this.get('currentUser.user');
+    user.get('skill');
+  },
```
Comment (JoshSmith) [inline app/routes/projects-list.js]: Don't know if this is necessary.

File: tests/integration/components/skill-list-item-test.js [hunk 3]
```diff
@@ -31,7 +31,7 @@ test('it renders and sends no action when not hidden', function(assert) {
   assert.expect(1);
 
   let skill = {
-    name: 'Ruby'
+    title: 'Ruby'
   };
```
Comment (JoshSmith) [inline tests/integration/components/skill-list-item-test.js]: I would actually love to rename this in the backend to `name`.

**Comment by JoshSmith** [comment]
@begedin I could use your thoughts now before I commit too much to this approach.

File: app/components/skill-list-item.js (hunk 1 above)
Comment (begedin) [inline app/components/skill-list-item.js]: Only other thing I could think of is to provide a callback to the sorting computed in the parent component. It's less weird, but it's also more work "at once", I think. Not sure which one would be quicker, though.

File: app/routes/projects-list.js (hunk 2 above)
Comment (hbrysiewicz) [inline app/routes/projects-list.js]: What made you think it was necessary? Removing this seems to be ok.. but curious what your concern was and why it was added.

Comment (JoshSmith) [inline app/routes/projects-list.js]: I thought that the user's skills were not loading explicitly from the user skills service.

File: app/services/user-skills.js [hunk 4]
```diff
@@ -23,6 +23,16 @@ export default Ember.Service.extend({
     return userSkill.save();
   },
 
+  hasSkill(skill) {
+    let userSkills = this.get('userSkills');
+    let matchedSkill = userSkills.find(function(item) {
+      let itemSkillId = item.belongsTo('skill').id();
+      let skillId = skill.get('id');
+      return (itemSkillId === skillId);
+    }.bind(this));
```
Comment (begedin) [inline app/services/user-skills.js]: Can ditch this bind in favor of an arrow function.

**Comment by begedin** [comment]
Looks good to me. You can decide on that bind vs arrow function thing.

**Pull request merged**